js/src/main/scala/org/soundsofscala/instrument/Instrument.scala

```diff
@@ -23,12 +23,23 @@ import org.soundsofscala
 import org.soundsofscala.models
 import org.soundsofscala.models.*
 
-trait Instrument:
+trait Instrument[Settings](using Default[Settings]):
 
-  def play(
+  final def play(
       musicEvent: AtomicMusicalEvent,
       when: Double,
       attack: Attack,
       release: Release,
-      tempo: Tempo)(using audioContext: dom.AudioContext): IO[Unit]
+      tempo: Tempo)(settings: Settings = Default.default[Settings])(using
+  audioContext: dom.AudioContext): IO[Unit] =
+    playWithSettings(musicEvent, when, attack, release, tempo, settings)
+
+  protected def playWithSettings(
+      musicEvent: AtomicMusicalEvent,
+      when: Double,
+      attack: Attack,
```

Comment:
At some point, maybe `attack`, `release`, etc. migrate into `Settings`? They have the same semantic role. Just a thought right now, not a suggestion that you make this change.

Reply:
Yes, definitely. I'll take care of it.